AirCasting/SessionViews/MeasurementsView.swift

```diff
@@ -0,0 +1,104 @@
+    
+    private var streamsToShow: [MeasurementStreamEntity] {
+        let allStreams = [session.pm1Stream,
+                          session.pm2Stream,
+                          session.pm10Stream,
+                          session.FStream,
+                          session.HStream]
+        
+        let toShow = allStreams.compactMap { $0 }
+        return toShow
+    }
+    private var hasAnyMeasurements: Bool {
+        streamsToShow.filter { $0.latestValue != nil }.count > 0
+    }
+    
+    var body: some View {
+        if hasAnyMeasurements {
+            VStack(alignment: .leading, spacing: 5) {
+                Text("Most recefnt measurement:")
+                HStack {
+                    Group {
+                        ForEach(streamsToShow) { stream in
+                            SingleMeasurementView(stream: stream,
+                                                  value: stream.latestValue ?? 0,
+                                                  threshold: threshold,
+                                                  selectedStream: _selectedStream)
+                        }
+                    }
+                    .frame(maxWidth: .infinity)
+                }
+            }
+        } else {
+            VStack(alignment: .leading, spacing: 3) {
+                Text("Your AirBeam is gathering data.")
+                    .font(Font.moderate(size: 14))
+                Text("Measurements will appear in 3 minutes.")
+                    .font(Font.moderate(size: 12))
+            }
+            .foregroundColor(.darkBlue)
+        }
+    }
+
+}
+
+struct SingleMeasurementView: View {
+    
+    let stream: MeasurementStreamEntity
+    let value: Double
+    var threshold: SensorThreshold
+    @Binding var selectedStream: MeasurementStreamEntity?
+    
+    var body: some View {
+        VStack(spacing: 3) {
+            #warning("Handle optional stream name")
+            Text(showStreamName())
+                .font(Font.system(size: 13))
+            
+            Button(action: {
+                selectedStream = stream
+            }, label: {
+                HStack(spacing: 3){
+                    MeasurementDotView(value: value,
+                                       thresholds: threshold)
+                    Text("\(Int(value))")
+                        .font(Font.moderate(size: 14, weight: .regular))
+                }
+            })
+            .buttonStyle(AirBorderedButtonStyle(isSelected: selectedStream == stream,
+                                                thresholdColor: colorBorder(stream: stream)))
+        }
+    }
+    
+    func showStreamName() -> String {
+        guard let streamName = stream.sensorName else {return ""}
+        return streamName
+            .drop { $0 != "-" }
+            .replacingOccurrences(of: "-", with: "")
+    }
+    
+    func colorBorder(stream: MeasurementStreamEntity) -> Color {
+        switch Int32(value) {
```

Comment:
there are multiple changes to compare trashholds and values into Int32 etc. Can we unify the types so no potential insecure projections would be made?

Reply:
Aren't those types imposed by the backend? I'm not sure what can I do about that.

Reply:
noting Is suggested by the backend. json only supports numeric type and has no limits to the values. Ofcourse as a separate task it makes no sense to separate it but worth considering because it can lead to very strange bugs in some corner cases. but should lt should be done sooner than later to avoid propagating the issue further.